src/main/java/chattime/command/MarkCommand.java

```diff
@@ -0,0 +1,50 @@
+package chattime.command;
+
+import chattime.TaskList;
+import chattime.storage.Storage;
+import chattime.task.Task;
+import chattime.ui.Ui;
+
+/**
+ * Represents MarkCommand object that handles main logic of marking a task as done or not done.
+ */
+public class MarkCommand extends Command {
+    private int taskIndex;
+    private boolean isDone;
+
+    /**
+     * Creates MarkCommand object to execute the task marking logic.
+     *
+     * @param index Index of the task in task list.
+     * @param taskIsDone true if mark as done, false if mark as undone.
+     */
+    public MarkCommand(int index, boolean taskIsDone) {
+        taskIndex = index;
+        isDone = taskIsDone;
+    }
+
+    /**
+     * Implements and executes main logic of MarkCommand object.
+     * Gets task from task list.
+     * Executes task status marking and returns UI message respectively to user.
+     * Updates data in storage file.
+     *  @param ui UI instance of bot.
```

Comment:
Consider leaving a single line separator at line 31 and removing the extra space precessing @param to ensure consistency of Javadoc formats across code files.

Reply:
Thanks for your reminder.